File: mobile_app/lib/models/hiring_request.dart
class HiringRequest {
  final String id;
  final String studentId;
  final String trainerId;
  final String status;
  final DateTime createdAt;
  final String? studentName;
  final String? studentPhoto;

  HiringRequest({
    required this.id,
    required this.studentId,
    required this.trainerId,
    required this.status,
    required this.createdAt,
    this.studentName,
    this.studentPhoto,
  });

  factory HiringRequest.fromJson(Map<String, dynamic> json) {
    return HiringRequest(
      id: json['id'].toString(),
      studentId: json['student_id'].toString(),
      trainerId: json['trainer_id'].toString(),
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
      studentName: json['student_name'],
      studentPhoto: json['student_photo'],
    );
  }
}

File: backend/app/models/hiring_request.py
import uuid
from datetime import datetime
import enum
from sqlalchemy import Column, String, DateTime, ForeignKey, Enum
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.db.session import Base

class HiringRequestStatus(str, enum.Enum):
    PENDING = "PENDING"
    ACCEPTED = "ACCEPTED"
    REJECTED = "REJECTED"

class HiringRequest(Base):
    __tablename__ = "hiring_requests"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    student_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=True)
    trainer_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    status = Column(Enum(HiringRequestStatus), default=HiringRequestStatus.PENDING, nullable=False)
    created_at = Column(DateTime, default=lambda: datetime.utcnow())
    updated_at = Column(DateTime, default=lambda: datetime.utcnow(), onupdate=lambda: datetime.utcnow())

    student = relationship("User", foreign_keys=[student_id], backref="sent_hiring_requests")
    trainer = relationship("User", foreign_keys=[trainer_id], backref="received_hiring_requests")
